internal/testrunner/runners/system/runner.go

```diff
@@ -629,11 +630,25 @@ func (r *runner) runTest(config *testConfig, ctxt servicedeployer.ServiceContext
 		hits, err = r.getDocs(dataStream)
 
 		if config.Assert.HitCount > 0 {
-			return hits.size() >= config.Assert.HitCount, err
-		}
+			if hits.size() == 0 {
+				return false, err
+			}
+
+			if hits.size() < config.Assert.HitCount {
+				return false, err
+			}
```

Comment:
Nit. First if is redundant.
```suggestion
			if hits.size() < config.Assert.HitCount {
				return false, err
			}
```